### PR TITLE
fix(bom): fix bom imports in camunda-only-bom  (#4628)

### PR DESCRIPTION
**Pull request by hauptmedia**
* use type pom with scope import for camunda-spin-bom
* use type pom with scope import for camunda-commons-bom
* use type pom with scope import for camunda-connect-bom
* remove project.version from dependency declarations
* use camunda-root as parent for camunda-spin-bom, camunda-commons-bom, camunda-connect-bom

related to https://github.com/camunda/camunda-bpm-platform/issues/4605

Backported commit 7644178 from the camunda-bpm-platform repository.
Original author: cachescrubber <lars.uffmann@gmail.com>

### CODE REVIEW

**Comment by hauptmedia** [comment]
This is the last commit that was missing from the c7 repo. It was kinda tricky to get this in because the pom files already changed a lot. So it would be nice if you could validate the changes @kthoms @javahippie

**Comment by kthoms** [comment]
I got https://github.com/operaton/operaton/issues/251 and since I observed this sometimes flaky, I have opened the issue.

However, with this change I got it now 3 times after each other. So far I could not dive into it, but I will try again. Maybe this change somehow uncovers it now reliable, but this is too early to say.

File: spin/pom.xml [hunk 1]
```diff
@@ -130,6 +131,15 @@
   <build>
     <pluginManagement>
       <plugins>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-javadoc-plugin</artifactId>
+          <version>${plugin.version.javadoc}</version>
```
Comment (kthoms) [inline spin/pom.xml:137]: plugin versions are managed by root /pom.xml. Remove this line.

**Review by kthoms — CHANGES_REQUESTED** [review]

File: commons/bom/pom.xml [hunk 2]
```diff
@@ -1,11 +1,13 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
 
+  <!-- HEADS-UP: Do not use operaton-parent as parent,
+  it would introduce a cyclic dependency to this bom -->
```
[the remark this answers is not in the document]
Comment (hauptmedia) [reply commons/bom/pom.xml]: removed it

File: spin/bom/pom.xml [hunk 3]
```diff
@@ -1,10 +1,12 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
 
+  <!-- HEADS-UP: Do not use operaton-parent as parent,
+  it would introduce a cyclic dependency to this bom -->
   <parent>
     <groupId>org.operaton.bpm</groupId>
```
[the remark this answers is not in the document]
Comment (hauptmedia) [reply spin/bom/pom.xml:5]: removed it

File: spin/pom.xml (hunk 1 above)
Comment (hauptmedia) [reply spin/pom.xml]: changed that

File: spin/pom.xml [hunk 4]
```diff
@@ -27,6 +27,7 @@
     <powermock.version>2.0.9</powermock.version>
     <mockito.version>3.3.3</mockito.version>
     <spin.version.old>1.23.0</spin.version.old>
+    <plugin.version.javadoc>3.3.1</plugin.version.javadoc>
```
[the remark this answers is not in the document]
Comment (hauptmedia) [reply spin/pom.xml]: changed that

**Comment by hauptmedia** [comment]
Thanks for reviewing! I integrated the changes, which can be inspected here: https://github.com/operaton/operaton/compare/a26f16d10eafc34d67508f38d870874de96bb76f..96dc1e88314ef8c66c5151ade9825de802723b2f

**Comment by kthoms** [comment]
Looks good. Let's wait for a successful build, but then should be fine.

**Review by kthoms — APPROVED** [review]

**Comment by hauptmedia** [comment]
Very nice, didn't expect to get this rather complex PR in on a Sunday evening, so we can have a fresh start on Monday. Thanks! 🥳

**Comment by kthoms** [comment]
Unfortunately broke something. I will fix that.

**Pull request merged**